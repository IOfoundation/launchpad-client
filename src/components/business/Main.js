import React, {Component} from 'react';
import {PropTypes} from 'prop-types';
import {browserHistory} from 'react-router';

import {MdShare, MdKeyboardBackspace} from 'react-icons/lib/md';
import ContentMap from '../layouts/ContentMap';

class Main extends Component {
  constructor(props) {
    super(props);
    this.reduceMap = this.reduceMap.bind(this);
    this.expandMap = this.expandMap.bind(this);
    this.state = {
      expanded: true,
    };
  }

  reduceMap() {
    this.setState({expanded: false});
  }

  expandMap() {
    this.setState({expanded: true});
  }
  _renderTopBar() {
    return (
      <div className="row between-xs middle-xs businessList_top businessTopBar">
        <div
          onClick={browserHistory.goBack}
          className="businessTopBar_btn middle-xs"
        >
          <MdKeyboardBackspace
            size={22}
            color={'#3F51B5'}
            className="businessTopBar_btn_icon"
          />
          <span>{'Back To Resources'}</span>
        </div>
        <div className="businessTopBar_btn middle-xs">
          <span className="businessTopBar_share">{'SHARE'}</span>
          <MdShare size={22} color={'#3F51B5'} />
        </div>
      </div>
    );
  }
  render() {
    const {business} = this.props;
    const businesses = [business];
    return (
      <ContentMap
        businesses={businesses}
        expanded={this.state.expanded}
        expandMap={this.expandMap}
        reduceMap={this.reduceMap}
        topBar={this._renderTopBar()}
      >
        <div className="bussinessView">
          <div>
            <img src={` ${business.logo}`} alt={`${business.name} logo`} />
          </div>
          <div>
            <h2>{business.name}</h2>
            <span>{'CONTACT'}</span>
            <ul>
              <li>{business.address}</li>
              <li
              >{`${business.city}, ${business.country_code} ${business.zip_code}`}</li>
            </ul>
            <ul>
              <li>{business.phone}</li>
              <li>{business.email}</li>
            </ul>
            <section className="row business_block--reduced_bottom">
              <a className="visitWebsite bold smallFont" href="#">
                {'VISIT WEBSITE'}
              </a>
              <a className="socialIcon" href="#">
                <img
                  className="socialIcon_icon"
                  src="../static-data/images/twitter.svg"
                />
              </a>
              <a className="socialIcon" href="#">
                <img
                  className="socialIcon_icon"
                  src="../static-data/images/facebook.svg"
                />
              </a>
              <a className="socialIcon" href="#">
                <img
                  className="socialIcon_icon"
                  src="../static-data/images/linkedin.svg"
                />
              </a>
            </section>
          </div>
          <div>
            <h3>{'SERVICE DESCRIPTION'}</h3>
            <p>{business.description}</p>
          </div>
          <div>
            <h3>{'SERVICES OFFERED'}</h3>
            <ul>
              {business.Services.map(service => (
                <li key={service.id} className={'services'}>{service.name}</li>
              ))}
            </ul>
          </div>
        </div>
      </ContentMap>
    );
  }
}

Main.propTypes = {
  business: PropTypes.object.isRequired,
};

export default Main;
