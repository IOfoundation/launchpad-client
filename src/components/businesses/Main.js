import React, {Component} from 'react';
import {PropTypes} from 'prop-types';

import BusinessesList from './BusinessesList';
import MapView from '../map-view/Main';
import Pagination from './Pagination';
import ResultInfo from './ResultInfo';

class Main extends Component {
  constructor(props) {
    super(props);
    this.state = {
      expanded: true,
    };
  }

  _reduceMap() {
    this.setState({expanded: false});
  }

  _expandMap() {
    this.setState({expanded: true});
  }
  _renderReduceButton() {
    return (
      <button className="reduceMapBtn" onClick={() => this._reduceMap()}>
        {'Reduce Map'}
      </button>
    );
  }
  _renderExpandButton() {
    return (
      <div className="row between-xs middle-xs reducedMapBottom">
        <button className="expandMapBtn" onClick={() => this._expandMap()}>
          {'Expand Map'}
        </button>
        <div className="reducedMapBottom_checkContainer">
          <input className="reducedMapBottom_check" type="checkbox" />
          <label className="reducedMapeBottom_checklabel">
            {'Update when map moves'}
          </label>
        </div>
      </div>
    );
  }
  _renderResultsInfo() {
    return <ResultInfo businessesMetadata={this.props.businessesMetadata} />;
  }
  render() {
    const {
      businessesMetadata,
      businesses,
      handleClickOnBusiness,
      handleChangePage,
    } = this.props;
    return (
      <div
        className={
          'businessesContainer ' +
          (this.state.expanded ? '' : 'businessesContainerReduced')
        }
      >
        {this.state.expanded ? null : this._renderResultsInfo()}
        <div
          className={
            'row ' + (this.state.expanded ? '' : 'businessesRow--reduced')
          }
        >
          <div
            className={
              this.state.expanded
                ? 'col-xs-5 businessList noPadding'
                : 'col-xs-8'
            }
          >
            {this.state.expanded ? this._renderResultsInfo() : null}
            <BusinessesList
              businesses={businesses}
              handleClickOnBusiness={handleClickOnBusiness}
              expanded={this.state.expanded}
            />
            <Pagination
              businessesMetadata={businessesMetadata}
              handleChangePage={handleChangePage}
            />
          </div>
          <div
            className={
              'map ' +
              (this.state.expanded
                ? 'col-xs-7 noPadding '
                : 'col-xs-4 mapReduced')
            }
          >
            <MapView businesses={businesses} />
            {this.state.expanded
              ? this._renderReduceButton()
              : this._renderExpandButton()}
          </div>
        </div>
      </div>
    );
  }
}

Main.propTypes = {
  businesses: PropTypes.array.isRequired,
  businessesMetadata: PropTypes.object.isRequired,
  handleChangePage: PropTypes.func.isRequired,
  handleClickOnBusiness: PropTypes.func.isRequired,
};

export default Main;
