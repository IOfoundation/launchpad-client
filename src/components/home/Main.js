import React from 'react';
import {PropTypes} from 'prop-types';
import {Link} from 'react-router';
import BusinessesForm from './BusinessesForm';

const Main = ({items, getTextSearchResults}) => {
  return (
    <div>
      <section className="hero center-xs">
        <div className="container-mid">
          <div className="center-x">
            <div className="remark">{'Resource Finder'}</div>
            <div className="col-lg-12">
              <h2 className="main-title desktop-devices text-bold">
                {'Where startups and small business connect in'}
                <br />
                {"California's Capital Region"}
              </h2>
              <h3 className="sub-title desktop-devices">
                {'Over 650 resources to help your business grow.'}
              </h3>
              <h2 className="mobile-devices text-bold">
                {
                  "Where startups and small business connect in California's Capital Region"
                }
              </h2>
              <div className="hero_inputContainer">
                <BusinessesForm
                  items={items}
                  getTextSearchResults={getTextSearchResults}
                />
              </div>
              <div className="margin-auto margin-y-20 text-thin">
                <Link
                  className="link hero-link p-bot-10 margin-x-16 m-bot-16"
                  to="/businesses?category=Planning/Management"
                >
                  {'Business Planning/Managment'}
                </Link>
                <Link
                  className="link hero-link p-bot-10 margin-x-16"
                  to="/businesses?category=Capital"
                >
                  {'Capital'}
                </Link>
                <Link
                  className="link hero-link p-bot-10 margin-x-16"
                  to="/businesses?category=Legal%20Services"
                >
                  {'Legal Services'}
                </Link>
                <Link
                  className="link hero-link p-bot-10 margin-x-16"
                  to="/businesses?category=Marketing/Sales"
                >
                  {'Marketing/Sales'}
                </Link>
                <Link
                  className="link hero-link p-bot-10 margin-x-16"
                  to="/businesses?category=Physical%20Space"
                >
                  {'Physical Space'}
                </Link>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  );
};

Main.propTypes = {
  getTextSearchResults: PropTypes.func,
  items: PropTypes.arrayOf(PropTypes.object),
  ResultsBusinesses: PropTypes.func,
};

export default Main;
