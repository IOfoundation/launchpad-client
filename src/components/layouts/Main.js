import React from 'react';
import {PropTypes} from 'prop-types';

import Header from '../shared/Header';
import Footer from '../shared/Footer';

const MainLayout = props => {
  return (
    <div>
      <Header />
      {props.children}
      <Footer />
    </div>
  );
};

MainLayout.propTypes = {};

export default MainLayout;
