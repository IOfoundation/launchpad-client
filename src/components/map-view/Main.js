import React, {Component} from 'react';
import {PropTypes} from 'prop-types';
import GoogleMap from 'google-map-react';
import MapMarker from './MapMarker';

class Main extends Component {
  constructor(props) {
    super(props);
    this.state = {
      selected: '-1',
    };
  }
  getCoordinates(business) {
    return business.coordinates;
  }
  handleBoundsChange(e) {
    this.props.onBoundsChange(e);
  }
  _handleOnClick(e) {
    this.setState({selected: e});
  }
  _handleCloseClick() {
    this.setState({selected: -1});
  }
  _handleChildMouseEnter(e) {
    this.props.highlightOrgCard(
      this.props.locations.find(x => String(x.id) === e).organization.id
    );
  }
  _handleChildMouseLeave() {
    this.props.highlightOrgCard(-1);
  }

  render() {
    const {locations} = this.props;
    const firstBusiness = locations ? locations[0] : null;
    const sacCoordinates = {lat: 38.57, lng: -121.47};
    const map_options = {fullscreenControl: false};
    const zoomLevel = 7;
    if (firstBusiness) {
      return (
        <GoogleMap
          center={sacCoordinates}
          zoom={zoomLevel}
          hoverDistance={12}
          onChange={e => this.handleBoundsChange(e)}
          resetBoundsOnResize={true}
          options={map_options}
          onChildMouseEnter={e => this._handleChildMouseEnter(e)}
          onChildMouseLeave={() => this._handleChildMouseLeave()}
          onChildClick={e => this._handleOnClick(e)}
          bootstrapURLKeys={{
            key: process.env.GOOGLE_MAP_API_KEY,
          }}
        >
          {locations.map(location => {
            const [lng, lat] = this.getCoordinates(location);
            return (
              <MapMarker
                key={location.id}
                lat={lat}
                lng={lng}
                organization={location.organization}
                selected={this.state.selected === String(location.id)}
                handleCloseClick={() => this._handleCloseClick()}
              />
            );
          })}
        </GoogleMap>
      );
    }
    return (
      <GoogleMap
        center={sacCoordinates}
        zoom={10}
        onChange={event => this.handleBoundsChange(event)}
        resetBoundsOnResize={true}
        options={map_options}
        bootstrapURLKeys={{
          key: process.env.GOOGLE_MAP_API_KEY,
        }}
      />
    );
  }
}

Main.propTypes = {
  highlightOrgCard: PropTypes.func.isRequired,
  locations: PropTypes.arrayOf(PropTypes.object),
  onBoundsChange: PropTypes.func.isRequired,
};

export default Main;
