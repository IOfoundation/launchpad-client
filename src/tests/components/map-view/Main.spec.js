import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import Main from 'components/map-view/Main';

const createProps = () => {
  return {
    expanded: false,
    highlightOrgCard: jest.fn(),
    isMobile: false,
    locations: [],
    onBoundsChange: jest.fn(),
    organizations: [],
    showLoading: false,
    toggleSwitch: false,
  };
};

describe('<Main />', () => {
  it('renders snapshot of map-view/Main', () => {
    const props = createProps();
    const wrapper = shallow(<Main {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
