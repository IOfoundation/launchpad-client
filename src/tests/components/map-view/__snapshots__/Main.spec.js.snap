// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> renders snapshot of map-view/Main 1`] = `
<p>
  There is no locations
</p>
`;
